### PR TITLE
[v12] refactor: exclude hidden elements by defaults

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary

- [x] Switch `defaultIncludeHiddenElements: false`
- [x] Update tests
- [x] Update docs

### Test plan

All checks should pass. Pls review testing code.

### CODE REVIEW

**Review by pierrezimmermannbam — APPROVED** [review]

File: src/helpers/accessiblity.ts [hunk 1]
```diff
@@ -59,6 +59,11 @@ export function isHiddenFromAccessibility(
 export const isInaccessible = isHiddenFromAccessibility;
 
 function isSubtreeInaccessible(element: ReactTestInstance): boolean {
+  // Null props can happen for React.Fragments
+  if (element.props == null) {
```
Comment (thymikee) [inline src/helpers/accessiblity.ts:63]: can it be undefined as well? if so, let's go with `===`

File: website/docs/API.md [hunk 2]
```diff
@@ -884,9 +884,7 @@ Default timeout, in ms, for async helper functions (`waitFor`, `waitForElementTo
 
 #### `defaultIncludeHiddenElements` option
 
-Default value for [includeHiddenElements](Queries.md#includehiddenelements-option) query option for all queries. Defaults to `true`, which means that queries will match [elements hidden from accessibility](#ishiddenfromaccessibility) by default.
-
-Currently this option is set to `true` which means that queries will also match hidden elements. This is done to avoid breaking changes. However, we plan to change the default behavior to exclude hidden elements in the next major release.
+Default value for [includeHiddenElements](Queries.md#includehiddenelements-option) query option for all queries. Defaults to `false`, which means that queries will not match [elements hidden from accessibility](#ishiddenfromaccessibility) by default.
```
Comment (thymikee) [inline website/docs/API.md:887]:
```suggestion
Default value for [includeHiddenElements](Queries.md#includehiddenelements-option) query option for all queries. Defaults to `false`, which means that queries will not match [elements hidden from accessibility](#ishiddenfromaccessibility) by default. Similar to the users that can't interact with those.
```

File: src/helpers/accessiblity.ts (hunk 1 above)
Comment (mdjastrzebski) [reply src/helpers/accessiblity.ts:63]: `== null` is a safer check, as it matches exactly `undefined` and `null`.

Comment (mdjastrzebski) [reply src/helpers/accessiblity.ts:63]: BTW according to official type info this cannot be `undefined` but is for `React.Fragments`

**Review by thymikee — APPROVED** [review]

File: src/helpers/accessiblity.ts (hunk 1 above)
Comment (thymikee) [reply src/helpers/accessiblity.ts:63]: Yeah this is what I asked for, just a bit indirectly. Let's leave it as is then

**Pull request merged**